src/gui/FileBrowser.cpp

```diff
@@ -121,11 +121,12 @@ FileBrowser::FileBrowser(const QString & directories, const QString & filter,
 	searchWidgetLayout->setContentsMargins(0, 0, 0, 0);
 	searchWidgetLayout->setSpacing( 0 );
 
-	m_filterEdit = new QLineEdit( searchWidget );
-	m_filterEdit->setPlaceholderText( tr("Search") );
-	m_filterEdit->setClearButtonEnabled( true );
-	connect( m_filterEdit, SIGNAL( textEdited( const QString& ) ),
-			this, SLOT( filterAndExpandItems( const QString& ) ) );
+	m_filterEdit = new QLineEdit(searchWidget);
+	m_filterEdit->setPlaceholderText(tr("Search"));
+	m_filterEdit->setClearButtonEnabled(true);
+	m_filterEdit->addAction(embed::getIconPixmap("zoom"), QLineEdit::LeadingPosition);
+	connect(m_filterEdit, SIGNAL(textEdited(const QString&)),
+			this, SLOT(filterAndExpandItems(const QString&)));
```

Comment:
This change does not compile for me:
```
/usr/include/qt/QtCore/qglobal.h:121:63: Fehler: statische Erklärung gescheitert: The slot requires more arguments than the signal provides.
```
Looks like this syntax does not understand the default parameters in `FileBrowser::filterAndExpandItems`.

Reply:
It's most likely because it's a recursive function, so it needs an extra parameter for the current item. I guess you can omit this one for now, but the other ones I mentioned should still apply.

Reply:
To fix this make the slot a wrapper function and recurse within a helper function.

Reply:
Starting with commit f8c81d5702a67b4bb330c9506f33702f7a64d136 a lambda with a matching signature is used to forward the call to the slot.